nevergrad/parametrization/data.py

```diff
@@ -0,0 +1,353 @@
+        self.exponent: t.Optional[float] = None
+        self.bounds: t.Tuple[t.Optional[np.ndarray], t.Optional[np.ndarray]] = (None, None)
+        self.bound_transform: t.Optional[trans.BoundTransform] = None
+        self.full_range_sampling = False
+
+    @property
+    def descriptors(self) -> core.Descriptors:
+        return core.Descriptors(deterministic=True, continuous=not self.integer)
+
+    def _get_name(self) -> str:
+        cls = self.__class__.__name__
+        descriptors: t.List[str] = ["int"] if self.integer else ([str(self.value.shape).replace(" ", "")] if self.value.shape != () else [])
+        descriptors += [f"exp={self.exponent}"] if self.exponent is not None else []
+        descriptors += [f"{self.bound_transform}"] if self.bound_transform is not None else []
+        descriptors += ["constr"] if self._constraint_checkers else []
+        description = ""
+        if descriptors:
+            description = "{{{}}}".format(",".join(descriptors))
+        return f"{cls}{description}"
+
+    @property
+    def sigma(self) -> t.Union["Array", "Scalar"]:
+        """Value for the standard deviation used to mutate the parameter
+        """
+        return self.subparameters["sigma"]  # type: ignore
+
+    @property
+    def value(self) -> np.ndarray:
+        if self.integer:
+            return np.round(self._value)  # type: ignore
+        return self._value
+
+    @value.setter
+    def value(self, value: np.ndarray) -> None:
+        if not isinstance(value, np.ndarray):
+            raise TypeError(f"Received a {type(value)} in place of a np.ndarray")
+        if self._value.shape != value.shape:
+            raise ValueError(f"Cannot set array of shape {self._value.shape} with value of shape {value.shape}")
+        if not BoundChecker(*self.bounds)(self.value):
+            raise ValueError("New value does not comply with bounds")
+        if self.exponent is not None and np.min(value.ravel()) <= 0:
+            raise ValueError("Logirithmic values cannot be negative")
+        self._value = value
+
+    def sample(self: A) -> A:
+        if not self.full_range_sampling:
+            return super().sample()
+        child = self.spawn_child()
+        std_bounds = tuple(self._to_std_space(b) for b in self.bounds)  # type: ignore
+        diff = std_bounds[1] - std_bounds[0]
+        child.set_standardized_data(std_bounds[0] + np.random.uniform(0, 1, size=diff.shape) * diff, deterministic=False)
+        return child
+
+    def set_bounds(self: A, a_min: BoundValue = None, a_max: BoundValue = None,
+                   method: str = "clipping", full_range_sampling: bool = False) -> A:
+        """Bounds all real values into [a_min, a_max] using a provided method
+
+        Parameters
+        ----------
+        a_min: float or None
+            minimum value
+        a_max: float or None
+            maximum value
+        method: str
+            "clipping", "constraint", "tanh" or "arctan"
```

Comment:
Setting bounds with "clipping" is quite clear to me, but I'm not sure if I understand what a clipping "constraint" or "tanh" is... Maybe adding an example can help?
Also the parameter `full_range_sampling` can be a little confusing. If I want to do uniform sampling, it seems more intuitive to have a function `param.uniform_sampling(minimum, maximum[, transform])`

Reply:
I've updated the description of `constraint` and `full_range_sampling`.
However having a method `uniform_sampling`  would not solve it: the aim is to have the algorithms use the `sample` when they need to create new instances from scratch. So this method must change behavior when we know that a new point should be created using a uniform sampling (so that the algorithm does not have to know anything about it + it does not make sense for any "non-data" parameter). Is it clearer? I tried to make this clearer in the description as well.
On the other hand, I don't like the name `full_range_sampling`, but I'm not sure using `uniform_sampling` would be better, since it's potentially **log-**uniform sampling sometimes as well. If you have a better name let me know :s